---
title: "manuscript"
format: 
  html:
    code-fold: true
editor: visual
---

```{r, echo=FALSE}
#| include = FALSE
library(tidyverse)

library("kableExtra")

```

```{r, echo=FALSE, cache=TRUE}
#| include = FALSE
tissue <- read_delim("data/salmon.merged.gene_tpm.tsv", col_names = TRUE)
```

## Introduction

The geoduck genome has 34,947 genes.

```{r schemat, echo = FALSE, out.width = "70%", fig.align = "center"}

knitr::include_graphics("img/genome.png")
```

```{r}
#| include = FALSE
gtab <- read_delim("data/20220419-pgen-gene-accessions-gene_id-gene_name-gene_description-alt_gene_description-go_ids.tab", col_names = FALSE)
```

```{r}
#| include = FALSE
head(gtab)
```

## Expression

```{r}
#| include = FALSE
top <- read_delim("data/geoduck_topmarkers_annot.txt", col_names = TRUE, na = "NA")
```


```{r}
#| include = FALSE
topwide <- top %>% 
select(gene_id, cell_group, specificity, gene_ID) %>%
pivot_wider(names_from = cell_group, values_from = specificity)

```


## Methods

## Results

Of the 34,947 genes, `r count(tissue %>% filter(ctenidia == 0, gonad == 0, heart == 0, juvenile == 0, larvae == 0))` were not expressed in any tissues. `r count(tissue %>% filter(ctenidia > 0, gonad > 0, heart > 0, juvenile > 0, larvae > 0))` genes were detected in all five tissues types. Table

```{r}
#| include = FALSE


count(tissue %>% filter(ctenidia > 0, gonad == 0, heart == 0, juvenile == 0, larvae == 0))

count(tissue %>% filter(ctenidia == 0, gonad > 0, heart == 0, juvenile == 0, larvae == 0))

count(tissue %>% filter(ctenidia == 0, gonad == 0, heart > 0, juvenile == 0, larvae == 0))

count(tissue %>% filter(ctenidia == 0, gonad == 0, heart == 0, juvenile > 0, larvae == 0))

count(tissue %>% filter(ctenidia == 0, gonad == 0, heart == 0, juvenile == 0, larvae > 0))
```

| Sample   | Unique Genes | Col3 |
|----------|--------------|------|
| ctenidia | 340          |      |
| gonad    | 119          |      |
| heart    | 371          |      |
| juvenile | 2180         |      |
| larvae   | 868          |      |

```{r}
#| include = FALSE
sc_num <- read_delim("data/rowData_df_genes_by_numcells_expressed", col_names = TRUE)
```

In addition to looking at expression profiles in tissue and whole organims (juvenile and larvae) single-cell sequencing was carried out developing embryo. In total `r count(sc_num %>% filter(num_cells_expressed > 0))` genes were expressed. Expressed varied as per Figure

```{r}
#| warning = FALSE
ggplot(sc_num, aes(x=num_cells_expressed)) + geom_histogram(bins = 100) +
scale_y_continuous(trans='log2')
```

As part of the single cell sequencing effort 14 distinct cell populations were identified with associated "top marker genes".


```{r}

topwide <- 
read_delim("data/geoduck_topmarkers_annot.txt", col_names = TRUE) %>% 
select(gene_id, cell_group, specificity, gene_ID) %>%
pivot_wider(names_from = cell_group, values_from = specificity) %>%
rename_with(~paste0("cellgroup_",.), -c("gene_id", "gene_ID"))

```

```{r}
knitr::kable(
topwide %>% filter(cellgroup_1 > 0) %>%
  select(gene_ID, cellgroup_1)

)
```





```{r}
#| warning = FALSE

kbl(
  left_join(topwide, tissue) %>%
  filter(cellgroup_1 >0) %>%
  select(gene_id, gene_ID, ctenidia, heart, gonad, larvae, juvenile) 
  , escape = F, digits=0, caption= "Cell Group 1 Top Markers") %>%
  kable_paper("striped", full_width = F)
```
```{r}
#| warning = FALSE

kbl(
  left_join(topwide, tissue) %>%
  filter(cellgroup_2 >0) %>%
  select(gene_id, gene_ID, ctenidia, heart, gonad, larvae, juvenile) 
  ,escape = F, digits=0, caption= "Cell Group 2 Top Markers") %>%
  kable_paper("striped", full_width = F)
```

```{r}
#| warning = FALSE

kbl(
  left_join(topwide, tissue) %>%
  filter(cellgroup_3 >0) %>%
  select(gene_id, gene_ID, ctenidia, heart, gonad, larvae, juvenile) 
  ,escape = F, digits=0, caption= "Cell Group 3 Top Markers") %>%
  kable_paper("striped", full_width = F)
```

```{r}
#| warning = FALSE

kbl(
  left_join(topwide, tissue) %>%
  filter(cellgroup_4 >0) %>%
  select(gene_id, gene_ID, ctenidia, heart, gonad, larvae, juvenile) 
  ,escape = F, digits=0, caption= "Cell Group 4 Top Markers") %>%
  kable_paper("striped", full_width = F)
```

```{r}
#| warning = FALSE

kbl(
  left_join(topwide, tissue) %>%
  filter(cellgroup_5 >0) %>%
  select(gene_id, gene_ID, ctenidia, heart, gonad, larvae, juvenile) 
  ,escape = F, digits=0, caption= "Cell Group 5 Top Markers") %>%
  kable_paper("striped", full_width = F)
```

```{r}
#| warning = FALSE

kbl(
  left_join(topwide, tissue) %>%
  filter(cellgroup_6 >0) %>%
  select(gene_id, gene_ID, ctenidia, heart, gonad, larvae, juvenile) 
  ,escape = F, digits=0, caption= "Cell Group 6 Top Markers") %>%
  kable_paper("striped", full_width = F)
```

```{r}
#| warning = FALSE

kbl(
  left_join(topwide, tissue) %>%
  filter(cellgroup_7 >0) %>%
  select(gene_id, gene_ID, ctenidia, heart, gonad, larvae, juvenile) 
  ,escape = F, digits=0, caption= "Cell Group 7 Top Markers") %>%
  kable_paper("striped", full_width = F)
```

```{r}
#| warning = FALSE

kbl(
  left_join(topwide, tissue) %>%
  filter(cellgroup_8 >0) %>%
  select(gene_id, gene_ID, ctenidia, heart, gonad, larvae, juvenile) 
  ,escape = F, digits=0, caption= "Cell Group 8 Top Markers") %>%
  kable_paper("striped", full_width = F)
```


```{r}
#| warning = FALSE

kbl(
  left_join(topwide, tissue) %>%
  filter(cellgroup_9 >0) %>%
  select(gene_id, gene_ID, ctenidia, heart, gonad, larvae, juvenile) 
  ,escape = F, digits=0, caption= "Cell Group 9 Top Markers") %>%
  kable_paper("striped", full_width = F)
```


```{r}
#| warning = FALSE

kbl(
  left_join(topwide, tissue) %>%
  filter(cellgroup_10 >0) %>%
  select(gene_id, gene_ID, ctenidia, heart, gonad, larvae, juvenile) 
  ,escape = F, digits=0, caption= "Cell Group 10 Top Markers") %>%
  kable_paper("striped", full_width = F)
```


```{r}
#| warning = FALSE

kbl(
  left_join(topwide, tissue) %>%
  filter(cellgroup_11 >0) %>%
  select(gene_id, gene_ID, ctenidia, heart, gonad, larvae, juvenile) 
  ,escape = F, digits=0, caption= "Cell Group 11 Top Markers") %>%
  kable_paper("striped", full_width = F)
```


```{r}
#| warning = FALSE

kbl(
  left_join(topwide, tissue) %>%
  filter(cellgroup_12 >0) %>%
  select(gene_id, gene_ID, ctenidia, heart, gonad, larvae, juvenile) 
  ,escape = F, digits=0, caption= "Cell Group 12 Top Markers") %>%
  kable_paper("striped", full_width = F)
```


```{r}
#| warning = FALSE

kbl(
  left_join(topwide, tissue) %>%
  filter(cellgroup_13 >0) %>%
  select(gene_id, gene_ID, ctenidia, heart, gonad, larvae, juvenile) 
  ,escape = F, digits=0, caption= "Cell Group 13 Top Markers") %>%
  kable_paper("striped", full_width = F)
```


```{r}
#| warning = FALSE

kbl(
  left_join(topwide, tissue) %>%
  filter(cellgroup_14 >0) %>%
  select(gene_id, gene_ID, ctenidia, heart, gonad, larvae, juvenile) 
  ,escape = F, digits=0, caption= "Cell Group 14 Top Markers") %>%
  kable_paper("striped", full_width = F)
```


